lib/logstash/inputs/kinesis.rb

```diff
@@ -49,8 +51,15 @@ class LogStash::Inputs::Kinesis < LogStash::Inputs::Base
   # to enable the cloudwatch integration in the Kinesis Client Library.
   config :metrics, :validate => [nil, "cloudwatch"], :default => nil
 
-  def initialize(params = {}, kcl_class = KCL::Worker)
-    @kcl_class = kcl_class
+  #what a nasty hack to use the overloaded method :_(
+  java_import KCL::Worker::Builder
+  class Builder
+    #I don't know how to make it work using the class directly
+    java_alias :v2RecordProcessorFactory, :recordProcessorFactory, [Java::JavaClass.for_name(KCL_PROCESSOR_FACTORY_CLASS)]
```

Comment:
can you explain this? i'm reading it as you are modifying the `KCL::Worker::Builder` class to alias `recordProcessorFactory` -> to `v2RecordProcessorFactory`, but i'm not clear on why